src/utils/linearAlg2D.h

```diff
@@ -119,8 +119,7 @@ class LinearAlg2D
         }
         else
         {
-            Point ret = p0 + projected_x / vSize(direction) * direction  / vSize(direction);
-            return ret;
+            return p0 + projected_x * direction / x_p1;
```

Comment:
Took me a minute to understand why it's the same! I'll add a bit of documentation here I think...